packages/core/src/node/env-variables/env-variables-server.ts

```diff
@@ -55,19 +55,20 @@ export class EnvVariablesServerImpl implements EnvVariablesServer {
         const dataFolderPath = join(BackendApplicationPath, 'data');
         const userDataPath = join(dataFolderPath, 'user-data');
         const dataFolderExists = this.pathExistenceCache[dataFolderPath] ??= await pathExists(dataFolderPath);
+        let theiaConfigDir: string;
         if (dataFolderExists) {
             const userDataExists = this.pathExistenceCache[userDataPath] ??= await pathExists(userDataPath);
             if (userDataExists) {
-                process.env.THEIA_CONFIG_DIR = userDataPath;
+                theiaConfigDir = userDataPath;
```

Comment:
I would move this assignment one level up, so eg to line 60 or even directly assign the theiaConfigDir to it and just override for the else case in line 69